src/Provider/Cache/ProviderCache.php

```diff
@@ -40,16 +40,24 @@ class ProviderCache implements Provider
      */
     protected $lifetime;
 
+    /**
+     * If true, include the real provider name into the cache key
+     *
+     * @var bool
+     */
+    protected $separateCache;
```

Comment:
Given that this package requires php ^7.4 this should be also:
```suggestion
    protected bool $separateCache;
```
Or maybe
```suggestion
    private bool $separateCache;
```
but that's not up to me :)

Reply:
As this syntax is not used anywhere in this repo, I did not dare.
(And also it was 3 years ago)

If I'm doing it, should I also update the 3 others properties in the same PR, or should we do that for every providers in an other PR?

Reply:
Ah you're right, I didn't though about that :facepalm: